pkg/operator/configobservation/apienablement/observe_runtime_config.go

```diff
@@ -17,8 +17,13 @@ import (
 
 var defaultGroupVersionsByFeatureGate = map[configv1.FeatureGateName][]groupVersionByOpenshiftVersion{
 	"MutatingAdmissionPolicy": {
-		{KubeVersionRange: semver.MustParseRange("< 1.34.0"), GroupVersion: schema.GroupVersion{Group: "admissionregistration.k8s.io", Version: "v1alpha1"}},
-		{KubeVersionRange: semver.MustParseRange(">= 1.34.0"), GroupVersion: schema.GroupVersion{Group: "admissionregistration.k8s.io", Version: "v1beta1"}},
+		// MutatingAdmissionPolicy was introduced as TechPreviewNoUpgrade in OCP 4.20 (k8s v1.33) with v1alpha1 API.
+		// OCP 4.21 (k8s v1.34) added v1beta1 API support alongside v1alpha1.
+		// Both APIs are expected to be removed when the feature goes GA.
+		// GA is tentatively assumed for k8s v1.35, but the KEP does not specify a version yet: https://github.com/kubernetes/enhancements/issues/3962
+		// TODO: Update these ranges when rebasing to k8s v1.35+
```

Comment:
To be specific: if we have a v1 version of the resources required for MutatingAdmissionPolicy in 1.35, then we can remove all references to MutatingAdmissionPolicy here. If we _don't_ get v1 resources, then these ranges should be bumped to "<1.36.0" and we'll make another decision during the following rebase.